src/main/java/org/optaplanner/springboottaskassigning/solver/SolverManager.java

```diff
@@ -16,16 +16,26 @@
 
 package org.optaplanner.springboottaskassigning.solver;
 
+import java.util.Optional;
+import java.util.function.Consumer;
+
 import org.optaplanner.core.api.score.Score;
+import org.optaplanner.core.api.solver.event.SolverEventListener;
 
 public interface SolverManager<Solution_> {
 
-    // TODO iterate over planning solution entities and set tenantId, or induce it from solution.getTenantId()
-    void solve(Comparable<?> tenantId, Solution_ planningSolution);
+    // FIXME CRITICAL BUG: iterate over planning solution entities and set tenantId, or induce it from solution.getProblemId()
+    void solve(Object problemId, Solution_ planningSolution,
+               Consumer<Solution_> onBestSolutionChangedEvent, Consumer<Solution_> onSolvingEnded);
+
+    // TODO add @Nullable annotation for getBestSolution, getBestScore and getSolverStatus
+    Solution_ getBestSolution(Object problemId);
+
+    Score getBestScore(Object problemId);
 
-    Solution_ getBestSolution(Comparable<?> tenantId);
+    SolverStatus getSolverStatus(Object problemId);
 
-    Score getBestScore(Comparable<?> tenantId);
+    void addEventListener(Object problemId, SolverEventListener<Solution_> eventListener);
```

Comment:
Do you need this addEventListener? when in doubt, leave it out - especially as it will be affected by the my prometeus related PR on optaplanner itself.

Reply:
To allow adding other eventListeners (the only other way is to add it when calling `solve()`). How does this affect prometheus PR?